build.gradle.kts

```
@@ -37,13 +48,34 @@ java {
 tasks.jar {
     manifest {
         attributes("Automatic-Module-Name" to "net.minecrell.terminalconsole")
+        attributes("Multi-Release" to "true")
+    }
+    into("META-INF/versions/11") {
+        from(sourceSets["java11"].output)
+        include("module-info.class")
     }
 }
 
+tasks.named<JavaCompile>("compileJava") {
+    options.release.set(8)
+}
+
+tasks.named<JavaCompile>("compileJava11Java") {
+    options.release.set(11)
+    options.javaModuleVersion.set(project.version as String)
+}
+
+tasks.named<JavaCompile>("compileIntTestJava") {
+    options.release.set(11)
+}
+
 tasks.withType<Test> {
     useJUnitPlatform()
 }
 
+val compileIntTestTask = tasks.named<JavaCompile>("compileIntTestJava").get()
+tasks.check { dependsOn(compileIntTestTask) }
```

```suggestion
tasks.check { dependsOn(tasks.named("compileIntTestJava")) }
```

Should be sufficient I think?